presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java

```diff
@@ -131,33 +130,46 @@ public static boolean isSystemSourcePartitioning(PartitioningHandle partitioning
         return false;
     }
 
+    public static OptionalInt getSystemPartitionCount(Partitioning partitioning)
+    {
+        ConnectorPartitioningHandle connectorHandle = partitioning.getHandle().getConnectorHandle();
+        if (connectorHandle instanceof SystemPartitioningHandle) {
+            return ((SystemPartitioningHandle) connectorHandle).getPartitionCount();
+        }
+        return OptionalInt.empty();
+    }
+
     private enum SystemPartitioning
     {
         FIXED,
         SOURCE,
         COORDINATOR_ONLY
     }
 
-    private static Partitioning createSystemPartitioning(SystemPartitioning partitioning, SystemPartitionFunction function)
+    private static Partitioning createSystemPartitioning(SystemPartitioning partitioning, SystemPartitionFunction function, OptionalInt partitionCount)
     {
-        return createSystemPartitioning(partitioning, function, ImmutableList.of());
+        return createSystemPartitioning(partitioning, function, partitionCount, ImmutableList.of());
     }
 
-    private static Partitioning createSystemPartitioning(SystemPartitioning partitioning, SystemPartitionFunction function, List<Symbol> columns)
+    private static Partitioning createSystemPartitioning(SystemPartitioning partitioning, SystemPartitionFunction function, OptionalInt partitionCount, List<Symbol> columns)
     {
-        return Partitioning.create(new PartitioningHandle(Optional.empty(), Optional.empty(), new SystemPartitioningHandle(partitioning, function)), columns);
+        return Partitioning.create(new PartitioningHandle(Optional.empty(), Optional.empty(), new SystemPartitioningHandle(partitioning, function, partitionCount)), columns);
     }
 
     private final SystemPartitioning partitioning;
     private final SystemPartitionFunction function;
+    private final OptionalInt partitionCount;
```

Comment:
we should clarify what an empty value here means. does it mean all? or irrelevant?